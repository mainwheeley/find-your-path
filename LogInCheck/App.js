import React, { Component } from 'react';
import { StyleSheet, View, Image, Text, KeyboardAvoidingView } from 'react-native';
import LoginForm from './comps/components/LoginForm';


export default class Login extends React.Component {
  render() {
    return (
      <KeyboardAvoidingView behavior="padding" style={styles.container}>
        <View style={styles.logoContainer}>
          <Image
            style={styles.logo}
            source={require('./comps/imgs/FYP.png')}
            />
            <Text style={styles.title}>An app made for github using React Native</Text>
        </View>
        <View style={styles.formContainer}>
          <LoginForm />
        </View>
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#8BC34A'
  },
  logoContainer: {
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'center'
  },
  logo: {
    width: 110,
    height: 100
  },
  title: {
    color: '#FFF',
    marginTop: 10,
    width: 160,
    textAlign: 'center',
    opacity: 0.9
  }
});
