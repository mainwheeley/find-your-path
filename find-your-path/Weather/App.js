import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View
} from 'react-native';

import api from './utilities/api.js';

export default class Weather extends Component {
  constructor(props){
    super(props);
    this.state = {
      weather: [],
      weatherTemp: '',
      weatherType: ''
    }
  }

  componentWillMount(){
    api.getWeathers().then((res) => {
      this.setState({
        weather: res.weather,
        weatherTemp: res.main.temp,
        weatherType: res.weather[0].description
      })
    });
  }

  render() {
    console.log("Weather: ", this.state.weather);
    console.log("Weather Temperature: ", this.state.weatherTemp);
    console.log("Weather Type: ", this.state.weatherType);
    return(
      <View style={styles.container}>
        <Text>
          Temperature in Celsius: {Math.round(this.state.weatherTemp - 273)}{"\n"}
          Temperature in Fahrenheit: {Math.round((this.state.weatherTemp - 273) * 9 / 5 + 32)} {"\n"}
          Temperature Description: {this.state.weatherType}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  }
})

AppRegistry.registerComponent('AwesomeProject', () => Weather);
