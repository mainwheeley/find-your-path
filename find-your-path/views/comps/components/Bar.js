import React from 'react';
import { StyleSheet, Dimensions, Text, View, Button, Alert, FlatList } from 'react-native';

export default class Bar extends React.Component {
   render() {
    return (

	<View style={styles.bar}>

	    <View style={[styles.barItem, styles.barseparator]}>
		<Text style={styles.barTop} onPress={() => this.props.navigation.navigate("FavoriteRoutes")}>Favorite Workouts</Text>
		<Text style={styles.barBottom}></Text>
	    </View>

	    <View style={[styles.barItem, styles.barseparator]}>
		<Text style={styles.barTop}>10</Text>
		<Text style={styles.barBottom}>Following</Text>
	    </View>

	</View>

    );
  }
}

const styles = StyleSheet.create({
    bar: {
	borderColor: '#fff',
	borderTopWidth: 4,
	height: Dimensions.get('window').height / 10,
	backgroundColor: '#8BC34A',
	flexDirection: 'row',
    },
    barSeparator: {
	borderRightWidth: 4
    },
    barTop: {
	color: '#fff',
	fontSize: 16,
	fontWeight: 'bold',
	fontStyle: 'italic',
    },
    barItem: {
	flex: 1,
	padding: 18,
	alignItems: 'center',
    },
    barBottom: {
	color: '#000',
	fontSize: 14,
	fontWeight: 'bold',
    },
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
