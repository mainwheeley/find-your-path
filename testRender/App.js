import React from 'react';

import {Text,View,TouchableOpacity, StyleSheet} from 'react-native';

export default class App extends React.Component {

constructor(){
    super()
    this.state = {
        viewSection :false
    }
}

renderBottomComponent(){
    if(this.state.viewSection) {
        return (
            <View>
                <Text>Hi!</Text>
            </View>
        )
    }
}

buttonPress=()=>{
    this.setState({viewSection:true})
}

render() {
    return (
        <View style={styles.container}>
            <TouchableOpacity onPress={this.buttonPress}>
                <Text> Click Me!</Text>
            </TouchableOpacity>
            {this.renderBottomComponent()}
        </View>
         );
      }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
