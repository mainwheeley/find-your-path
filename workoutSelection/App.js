import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import RadioForm, {RadioButton, RadioButtonInput, RadioButtonLabel} from 'react-native-simple-radio-button';

var radio_props = [
  {label: 'Walking', value: 'walk' },
  {label: 'Running', value: 'run' },
  {label: 'Bicycle', value: 'bike' },
  {label: 'Other', value: 'etc' },

];

export default class App extends React.Component {
    constructor() {
	super();
	this.state = {
	    value: 'walk'
	};
    }

    logOnChange(valuex) {
	this.setState({value:valuex});
	console.log(valuex);
    }

  render() {
    return (
      <View style={styles.container}>
	<RadioForm
	    radio_props={radio_props}
	    initial={0}
	    onPress={(value) => 
		{this.logOnChange(value)}
	    }
	    buttonColor={'#8BC34A'}
	    formHorizontal={true}
	    labelHorizontal={false}
	/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
