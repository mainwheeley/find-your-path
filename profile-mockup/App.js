import React from 'react';
import { 
    StyleSheet,
    Text,
    View,
    Button,
    Alert,
    FlatList,
    Image,
    ScrollView
} from 'react-native';

import Header from './comps/components/Header';
import buttontest from './comps/components/buttontest';
import Bar from './comps/components/Bar';
import Stats from './comps/components/Stats';

export default class App extends React.Component {
    alertMess() {
	Alert.alert("Settings button pressed");
    }

    itemSeparator = () => (
	<View
	    style={{
		backgroundColor: '#000',
		height: 2,
	    }}
	/>
    );

    render() {
    return (
 //   <View style = {styles.container}>
	<ScrollView style={styles.container}> 

	    <Header />
	    <Bar />
	    <Stats />

	</ScrollView>
//    </View>

//	<View style={styles.container}>	
//	    <FlatList
//		data={[{key: 'Total running time: '}, {key: 'Total miles run'}]}
//  	    	renderItem={({item}) => <Text style={styles.statStyle}>{item.key}</Text>}
//		ItemSeparatorComponent={this.itemSeparator}
//	    />
//	</View>
    );
  }
}

const styles = StyleSheet.create({
    statStyle: {
	marginTop: 20,
	marginBottom: 20,
	fontSize: 16,
	color: '#000',
	fontWeight: 'bold'
    },
    container: {
	flex: 1,
	backgroundColor: '#fff',
    }
});

//var test = React.createClass
