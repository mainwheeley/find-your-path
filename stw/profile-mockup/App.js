import React from 'react';
import { 
    StyleSheet,
    TouchableOpacity,
    Dimensions,
    Text,
    TextInput,
    View,
    Button,
    Alert,
    FlatList,
    Image,
    ScrollView
} from 'react-native';

import Header from './comps/components/Header';
import buttontest from './comps/components/buttontest';
import Bar from './comps/components/Bar';
import Stats from './comps/components/Stats';
import deleteAcc from './comps/components/deleteAcc';

export default class App extends React.Component {
    constructor() {
	super();
    	this.state = { 
	    text: 'Useless Placeholder',
	    viewSection: false
	};
    }

    itemSeparator = () => (
	<View
	    style={{
		backgroundColor: '#000',
		height: 2,
	    }}
	/>
    );

    renderStats() {
	if (!this.state.viewSection) {
	    return (
		<Stats />
	    )
	}
    }

    setStatsRender=()=> {
	this.setState({viewSection: true})
    }

    render() {
    return (
 //   <View style = {styles.container}>
	<View style={styles.container}> 

	    <Header />
	    <Bar />
	    <TouchableOpacity onPress={this.setStatsRender}>
		<Text> Render Stats!</Text>
	    </TouchableOpacity>
	    {this.renderStats()}

	</View>
//    </View>

//	<View style={styles.container}>	
//	    <FlatList
//		data={[{key: 'Total running time: '}, {key: 'Total miles run'}]}
//  	    	renderItem={({item}) => <Text style={styles.statStyle}>{item.key}</Text>}
//		ItemSeparatorComponent={this.itemSeparator}
//	    />
//	</View>
    );
  }
}

const styles = StyleSheet.create({
    statStyle: {
	marginTop: 20,
	marginBottom: 20,
	fontSize: 16,
	color: '#000',
	fontWeight: 'bold'
    },
    container: {
	flex: 1,
	backgroundColor: '#fff',
    }
});

//var test = React.createClass
