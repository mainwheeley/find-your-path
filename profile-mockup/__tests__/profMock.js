import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import App from '../App';
import Bar from '../comps/components/Bar';
import { shallow } from 'enzyme';
describe('Profile Page', function() {
    it('renders correctly', () => {
	const tree = shallow(<App />)
	expect(tree).toMatchSnapshot();
    });
    it('length 1', function() {
	const tree = shallow(<App />)
	expect(tree.length).toBe(1);
    });
});
