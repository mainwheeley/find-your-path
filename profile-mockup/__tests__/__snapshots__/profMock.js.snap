// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <Header />
    <Bar />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": 0,
          "borderColor": "#fff",
          "borderWidth": 5,
          "flexDirection": "row",
          "height": 133.4,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#F44336",
            "borderLeftWidth": 5,
            "borderRadius": 10,
            "borderRightWidth": 5,
            "borderWidth": 1,
            "flex": 1,
            "height": 88.93333333333334,
            "width": 88.93333333333334,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 78.47058823529412,
              "width": 78.47058823529412,
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#F44336",
            "borderLeftWidth": 5,
            "borderRadius": 10,
            "borderRightWidth": 5,
            "borderWidth": 1,
            "flex": 1,
            "height": 88.93333333333334,
            "width": 88.93333333333334,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 78.47058823529412,
              "width": 78.47058823529412,
            }
          }
        />
      </TouchableOpacity>
    </View>
    <Stats />
  </View>,
  "nodes": Array [
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
        }
      }
    >
      <Header />
      <Bar />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": 0,
            "borderColor": "#fff",
            "borderWidth": 5,
            "flexDirection": "row",
            "height": 133.4,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#F44336",
              "borderLeftWidth": 5,
              "borderRadius": 10,
              "borderRightWidth": 5,
              "borderWidth": 1,
              "flex": 1,
              "height": 88.93333333333334,
              "width": 88.93333333333334,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 78.47058823529412,
                "width": 78.47058823529412,
              }
            }
          />
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#F44336",
              "borderLeftWidth": 5,
              "borderRadius": 10,
              "borderRightWidth": 5,
              "borderWidth": 1,
              "flex": 1,
              "height": 88.93333333333334,
              "width": 88.93333333333334,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 78.47058823529412,
                "width": 78.47058823529412,
              }
            }
          />
        </TouchableOpacity>
      </View>
      <Stats />
    </View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <App />,
    "_instance": App {
      "context": Object {},
      "itemSeparator": [Function],
      "props": Object {},
      "refs": Object {},
      "setStatsRender": [Function],
      "setWorkoutsRender": [Function],
      "state": Object {
        "text": "Useless Placeholder",
        "viewStats": true,
        "viewWorkouts": false,
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
        }
      }
    >
      <Header />
      <Bar />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderBottomWidth": 0,
            "borderColor": "#fff",
            "borderWidth": 5,
            "flexDirection": "row",
            "height": 133.4,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#F44336",
              "borderLeftWidth": 5,
              "borderRadius": 10,
              "borderRightWidth": 5,
              "borderWidth": 1,
              "flex": 1,
              "height": 88.93333333333334,
              "width": 88.93333333333334,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 78.47058823529412,
                "width": 78.47058823529412,
              }
            }
          />
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#F44336",
              "borderLeftWidth": 5,
              "borderRadius": 10,
              "borderRightWidth": 5,
              "borderWidth": 1,
              "flex": 1,
              "height": 88.93333333333334,
              "width": 88.93333333333334,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 78.47058823529412,
                "width": 78.47058823529412,
              }
            }
          />
        </TouchableOpacity>
      </View>
      <Stats />
    </View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
